src/core/UserIconGroup/UserIconGroup.tsx

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import cx from 'classnames';
+import { UserIcon } from '../UserIcon';
+export type UserIconGroupType = 'default' | 'animated';
+
+const defaultLength = 5;
+const maxLength = 9;
+
+export type UserIconGroupProps = {
+  /**
+   * UserIconGroup group length
+   * @default 5
+   */
+  length?: number;
+  /**
+   * UserIconGroup type
+   * @default 'default'
+   */
+  type?: UserIconGroupType;
+  /**
+   * User Icons in the UserIconGroup.
+   */
+  children: React.ReactNode;
+};
+
+/**
+ * Group User Icons together
+ * @example
+ * <UserIconGroup>
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *   abbreviation="RM"
+ *    backgroundColor="#C8C2B4"
+ *    size="medium"
+ *    title="Robin Mercer"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ * </UserIconGroup>
+ * * @example
+ * <UserIconGroup type='animated'>
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *   abbreviation="RM"
+ *    backgroundColor="#C8C2B4"
+ *    size="medium"
+ *    title="Robin Mercer"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ * </UserIconGroup>
+ */
+export const UserIconGroup = (props: UserIconGroupProps) => {
+  const { children, type = 'default', length = defaultLength } = props;
+
+  const childrenArray = React.Children.toArray(children);
+  const childrenLength = React.Children.count(children);
+  let additionalChildren = '';
+  if (childrenLength > length) {
+    React.Children.forEach(children, (child: JSX.Element, index) => {
+      if (index >= length) {
+        additionalChildren += child.props.title + '\n';
+      }
+    });
+  }
+
+  useTheme();
+
+  return (
+    <div
+      className={cx('iui-user-icon-list', {
+        'iui-animated': type == 'animated',
+      })}
+    >
+      {childrenLength <= length ? (
+        children
+      ) : (
+        <>
+          {childrenArray.map((child, index) =>
+            index < length
+              ? React.cloneElement(child as JSX.Element, {
+                  status: '',
+                  key: index,
+                })
+              : null,
+          )}
+
+          <UserIcon
```

Comment:
Extra members icon color: 
orange in https://itwin.github.io/iTwinUI/backstop/minified/user-icon.html
![image](https://user-images.githubusercontent.com/83585998/124271604-54b0d200-db46-11eb-99f2-4adbba5c56f8.png)

gray in https://ux.bentley.com/itwin/components/user-icon.html#team-displays
![image](https://user-images.githubusercontent.com/83585998/124271638-5d090d00-db46-11eb-9de8-a6a91f9bd82c.png)

white when no "background-color" prop passed
![image](https://user-images.githubusercontent.com/83585998/124271795-917cc900-db46-11eb-8922-157ed1baf106.png)

green when empty string as "background-color" prop passed
![image](https://user-images.githubusercontent.com/83585998/124271759-86299d80-db46-11eb-90b5-4ad4bb30b265.png)

Reply:
@FlyersPh9

Reply:
Created a new PR to give the user icon with the count inside it slightly different visual styling: https://github.com/iTwin/iTwinUI/pull/177